guides/asynchronous-tasks/readme.md

```diff
@@ -183,22 +183,24 @@ barrier.wait
 
 ## Stopping a Task
 
-When a task completes execution, it will enter the `complete` state (or the `failed` state if it raises an unhandled exception).
```

Comment:
All exceptions will cause the task to enter the failed state.

https://github.com/socketry/async/blob/main/lib/async/task.rb#L314-L317

Reply:
I thought you were saying that non-`StandardError`-derived exceptions would shut down the reactor and cause it to re-`raise`?

Reply:
Yes, the task will still enter the failed state though in any case where an exception propagates up to the root of the fiber running the task. I suppose your change is okay because it's just mentioning one case, but both cases are true - i.e.:

> (or the `failed` state if it raises an unhandled `StandardError`-derived exception or any exception)

But I think we can just say "raises an unhandled exception" since in every case the task enters the `failed` state.

Reply:
(If you think it's confusing, I'm also okay with your change, but it should be noted that any exception that propagates up causes the task to enter a `failed` state.)